File: vdp-ruby/test/visacheckout/update_payment_data_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class UpdatePaymentDataTest < Test::Unit::TestCase
  
  def setup
    @config = YAML.load_file('configuration.yml')
    @visa_api_client = VisaAPIClient.new
    @updatePaymentInfoRequest = '''{
      "orderInfo": {
      "currencyCode": "USD",
      "discount": "5.25",
      "eventType": "Confirm",
      "giftWrap": "10.1",
      "misc": "3.2",
      "orderId": "testorderID",
      "promoCode": "testPromoCode",
      "reason": "Order Successfully Created",
      "shippingHandling": "5.1",
      "subtotal": "80.1",
      "tax": "7.1",
      "total": "101"
      }
   }'''
  end
  
  def test_updatePaymentInfo
    base_uri = 'wallet-services-web/'
    resource_path = 'payment/info/{callId}'
    resource_path = resource_path.sub('{callId}',@config['checkoutCallId'])
    api_key = @config['apiKey']
    response_code = @visa_api_client.doXPayTokenRequest(base_uri, resource_path, "apikey=#{api_key}", "Update Payment Data Test", "put", @updatePaymentInfoRequest)
    assert_equal("200", response_code, "Update Payment Information test failed")
    end
end

File: visa-dev-samplecode-master/vdp-node/test/visacheckout/updatepaymentdatatest.js
var request = require('request');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var config = require('../../config/configuration.json');
var assert = require('chai').assert;
var req = request.defaults();
var randomstring = require('randomstring');

describe('Update Payment Information', function() {

	var visaAPIClient = new VisaAPIClient();
	var updatePaymentInfoRequest = JSON.stringify({
        "orderInfo": {
            "currencyCode": "USD",
            "discount": "5.25",
            "eventType": "Confirm",
            "giftWrap": "10.1",
            "misc": "3.2",
            "orderId": "testorderID",
            "promoCode": "testPromoCode",
            "reason": "Order Successfully Created",
            "shippingHandling": "5.1",
            "subtotal": "80.1",
            "tax": "7.1",
            "total": "101"
          }
       });
	
	it('Update Payment Data Test',function(done){
		this.timeout(10000);
		var apiKey = config.apiKey;
		var baseUri = 'wallet-services-web/';
		var resourcePath = 'payment/info/{callId}';
		resourcePath = resourcePath.replace('{callId}', config.checkoutCallId);
		var queryParams = 'apikey=' + apiKey;
		visaAPIClient.doXPayRequest(baseUri, resourcePath, queryParams, updatePaymentInfoRequest, 'PUT', {}, 
		function(err, responseCode) {
			if(!err) {
				assert.equal(responseCode, 200);
			} else {
				assert(false);
			}
			done();
		});
	});
});
